ui/src/app/base/hooks/node.ts

```diff
@@ -112,3 +114,23 @@ export const useCanEdit = (
     (ignoreRackControllerConnection || isRackControllerConnected)
   );
 };
+
+/**
+ * Return a list of tags that belong to a given node.
+ * @param node - The node whose tags to retrieve.
+ * @returns A list of tags that belong to the node.
+ */
+export const useNodeTags = (node: Node | null): Tag[] => {
+  const tags = useSelector(tagSelectors.all);
+
+  if (!node) {
+    return [];
+  }
+  return node.tags.reduce<Tag[]>((nodeTags, tagID) => {
```

Comment:
I wonder if there's any need for this to be specific to tags? It could just take an array of tag ids. In that case it could be done through a tag selector that takes multiple tag ids and then we'd get memoize benefits as well.

Reply:
That's not a bad idea - I might switch to using a tag selector instead